Base/res/js/Spreadsheet/runtime.js

```diff
@@ -365,7 +365,9 @@ function randRange(min, max) {
 }
 
 function integer(value) {
-    return value | 0;
+    // Can check for symbol too, but it's not needed.
+    const asNum = value | 0;
+    return asNum === 0 ? (value == "0" ? 0 : NaN) : asNum;
```

Comment:
NaN is not a valid value for an integer.

Reply:
fixed, it now throws in weird cases :)